dao-voting/src/main.sw

```diff
@@ -0,0 +1,272 @@
+contract;
+
+dep abi;
+dep data_structures;
+dep errors;
+
+use std::{
+    address::Address,
+    assert::require,
+    block::height,
+    chain::auth::{AuthError, msg_sender},
+    context::{call_frames::msg_asset_id, msg_amount, this_balance},
+    contract_id::ContractId,
+    identity::Identity,
+    result::*,
+    revert::revert,
+    storage::StorageMap,
+    token::{transfer_to_output, force_transfer_to_contract},
+};
+
+use abi::DaoVoting;
+use data_structures::{CallData, Proposal};
+use errors::Error;
+
+storage {
+    gov_token: ContractId,
+    voting_period: u64,
+    approval_percentage: u64,
+    proposals: StorageMap<u64,
+    Proposal>, proposal_count: u64,
+    // The amount of governance tokens a user has deposited
+    balances: StorageMap<Identity,
+    u64>,
+    // The amount of votes a user has used on a proposal
+    votes: StorageMap<(Identity, u64),
+    u64>, state: u64, 
+}
+
+impl DaoVoting for Contract {
+    /// Initialize the dao with the governance token, voting parameters, and the proposal.
+    ///
+    /// # Panics
```

Comment:
Take a look at @Braqzen's fundraiser pull request and try to match this format. The documentation should include arguments